src/core/solver.py

```
@@ -344,20 +576,37 @@ def _is_missing(value: object) -> bool:
+        float: Coerced float value.
+    """
     if _is_missing(value):
         return 0.0
     if isinstance(value, str) and not value.strip():
         return 0.0
     try:
-        return float(value)
+        val = float(value)
+        if math.isnan(val) or math.isinf(val):
+            return 0.0
+        return val
```

Ruff (0.15.12)

[warning] Consider moving this statement to an `else` block

(TRY300)